### PR TITLE
Migrate jobs to use --provider, --gcp-project, --gcp-zone, --gcp-cloud-sdk and off KUBEMARK_TESTS

### PR DESCRIPTION
**Pull request by fejta**
/assign @krzyzacy @yguo0905 
/cc @zmerlynn @gmarek 

`KUBEMARK_TESTS=foo` is equivalent to `--test_args=--ginkgo.focus=foo`
Other envs now have an equivalent kubetest flag.
Only migrating kubernetes_e2e.py scenario jobs for now (skipping kops)

ref https://github.com/kubernetes/test-infra/issues/3375

### CODE REVIEW

File: jobs/ci-kubernetes-e2e-gke-ubuntustable1-k8sstable1-alphafeatures.env [hunk 1]
```diff
@@ -3,11 +3,9 @@
 # The common configurations.
 
 # The cloud provider configurations.
-KUBERNETES_PROVIDER=gke
 E2E_MIN_STARTUP_PODS=8
 ZONE=us-central1-f
```
Comment (yguo0905) [inline jobs/ci-kubernetes-e2e-gke-ubuntustable1-k8sstable1-alphafeatures.env]: Remove this here and for the other gke-ubuntustable1 jobs? `bazel run //experiment:generate_tests -- --yaml-config-path=experiment/test_config.yaml --json-config-path=jobs/config.json --prow-config-path=prow/config.yaml && bazel run //jobs:config_sort` should work.

File: jobs/config.json [hunk 2]
```diff
@@ -562,7 +591,8 @@
       "--extract=ci/latest-1.7",
       "--timeout=180m",
       "--check-leaked-resources",
-      "--test_args=--ginkgo.focus=\\[Feature:(ExternalTrafficLocalOnly|DynamicKubeletConfig)\\]|Initializers"
+      "--test_args=--ginkgo.focus=\\[Feature:(ExternalTrafficLocalOnly|DynamicKubeletConfig)\\]|Initializers",
+      "--gcp-project=k8s-jkns-e2e-gce-serial-1-2"
```
Comment (krzyzacy) [inline jobs/config.json]: --provider=gce? and other places

File: jenkins/bootstrap_test.py [hunk 3]
```diff
@@ -2200,18 +2201,23 @@ def test_envs_no_export(self):
                     ('JENKINS_USE_SERVER_VERSION=', '--extract=gke'),
                     ('JENKINS_USE_GCI_VERSION=', '--extract=gci/FAMILY'),
                     ('JENKINS_USE_GCI_HEAD_IMAGE_FAMILY=', '--extract=gci/FAMILY'),
+                    ('KUBE_GCE_ZONE=', '--gcp-zone='),
                     ('KUBEKINS_TIMEOUT=', '--timeout=XXm'),
+                    ('KUBEMARK_TESTS=', '--test_args=--ginkgo.focus=FOO'),
+                    ('KUBERNETES_PROVIDER=', '--provider=FOO'),
                     ('PERF_TESTS=', '--perf'),
+                    ('PROJECT=', '--gcp-project=FOO'),
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: there's a test to check duplicated projects per job, please also update there.

File: jobs/config.json [hunk 4]
```diff
@@ -1970,7 +2165,8 @@
       "--timeout=180m",
       "--extract=ci/latest-1.7",
       "--check-leaked-resources",
-      "--test_args=--ginkgo.focus=\\[Feature:GPU\\]"
+      "--test_args=--ginkgo.focus=\\[Feature:GPU\\]",
+      "--gcp-project=k8s-jkns-e2e-gce-gpus-1-7"
```
Comment (mindprince) [inline jobs/config.json]: Add
```
      "--gcp-zone=us-west1-b"
```
here.

File: jobs/config.json [hunk 5]
```diff
@@ -5296,11 +6076,15 @@
   "ci-kubernetes-e2e-gke-gpu": {
     "args": [
       "--env-file=jobs/platform/gke.env",
+      "--provider=gke",
+      "--gcp-zone=us-central1-f",
```
Comment (mindprince) [inline jobs/config.json]: The env file for this specifies:
```
 ZONE=us-west1-b
```

File: jobs/config.json (hunk 4 above)
Comment (mindprince) [inline jobs/config.json]: Also, we can delete `"--env-file=jobs/ci-kubernetes-e2e-gce-gpu-1-7.env",` from this block, now that the file doesn't is empty.

Comment (mindprince) [inline jobs/config.json]: Actually there's a test which check for the existence of this flag, so don't remove it.

File: jobs/config.json (hunk 5 above)
Comment (fejta) [inline jobs/config.json]: Done

File: jobs/config.json (hunk 4 above)
Comment (fejta) [inline jobs/config.json]: Done

File: jobs/config.json (hunk 2 above)
Comment (fejta) [inline jobs/config.json]: Done

File: jobs/ci-kubernetes-e2e-gke-ubuntustable1-k8sstable1-alphafeatures.env (hunk 1 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gke-ubuntustable1-k8sstable1-alphafeatures.env]: Done

File: jobs/config.json [hunk 6]
```diff
@@ -7701,7 +9364,12 @@
   },
   "pull-kubernetes-e2e-gke": {
     "args": [
+      "--gcp-project=k8s-jkns-pr-gke",
+      "--gcp-zone=us-central1-f",
       "--env-file=jobs/platform/gke.env",
+      "--provider=gke",
+      "--gcp-cloud-sdk=gs://cloud-sdk-testing/ci/staging",
+      "--gcp-zone=us-west1-b",
```
Comment (krzyzacy) [inline jobs/config.json]: seems there can be multiple `--gcp-zone=`, only keep one?

File: jobs/config.json [hunk 7]
```diff
@@ -5383,9 +6568,13 @@
     "args": [
       "--env-file=jobs/platform/gke.env",
       "--provider=gke",
+      "--gcp-cloud-sdk=gs://cloud-sdk-testing/ci/staging",
+      "--provider=gke",
       "--gcp-zone=us-west1-b",
       "--gcp-cloud-sdk=gs://cloud-sdk-testing/ci/staging",
       "--env-file=jobs/ci-kubernetes-e2e-gke-gpu.env",
+      "--gcp-project=k8s-jkns-e2e-gke-gpus",
+      "--gcp-zone=us-west1-b",
```
Comment (mindprince) [inline jobs/config.json]: This is a newly added job (added couple of hours ago), I added it using the new format already, so these flags are duplicate now. And one of them (`--gcp-project`) is incorrect.

File: jobs/config.json [hunk 8]
```diff
@@ -1980,8 +2318,12 @@
   },
   "ci-kubernetes-e2e-gce-gpu-1-7": {
     "args": [
+      "--gcp-project=k8s-jkns-e2e-gce-gpus-1-7",
       "--env-file=jobs/platform/gce.env",
+      "--provider=gce",
       "--env-file=jobs/ci-kubernetes-e2e-gce-gpu.env",
+      "--gcp-project=k8s-jkns-e2e-gce-gpus",
```
Comment (mindprince) [inline jobs/config.json]: This flag occurs twice and this one is wrong.

**Comment by krzyzacy** [comment]
ok, my biggest potential issue is, there are still places we are using those envs, in main repo, for example, https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/list-resources.sh, and other thousands of bash goodies. Maybe we should still set the envs back to make it backwards compatible for now.

**Comment by fejta** [comment]
Kubetest sets these variables

**Comment by krzyzacy** [comment]
ah, yes `migratedOption` handled that, ack.

File: jobs/config.json (hunk 7 above)
Comment (fejta) [inline jobs/config.json]: Done

File: jobs/config.json (hunk 8 above)
Comment (fejta) [inline jobs/config.json]: Done

File: jobs/config.json (hunk 6 above)
Comment (fejta) [inline jobs/config.json]: Done

**Comment by fejta** [comment]
PTAL!

This version of move_extract.py has a lot of grossness :grin:

File: jenkins/check_projects.py [hunk 9]
```diff
@@ -145,7 +145,7 @@ def parse_args():
     parser = argparse.ArgumentParser(
         description=__doc__, formatter_class=RawTextHelpFormatter)
     parser.add_argument(
-        '--filter', default=r'^.+\.(env|sh)$',
+        '--filter', default=r'^.+$',
         help='Only look for projects with the specified names')
```
Comment (krzyzacy) [inline jenkins/check_projects.py]: omg... I love the fact that this help msg is correct now

**Comment by fejta** [comment]
Okay should pass lint now

**Comment by mindprince** [comment]
The GPU changes look good to me now.

**Comment by krzyzacy** [comment]
/lgtm

maybe consider bikeshed `us-central-1` as the default zone?

**Comment by fejta** [comment]
Hmm, I guess we should shove that into kubernetes_e2e.py? Will send a PR to do this soon.

**Comment by fejta** [comment]
fyi @spxtr

**Pull request merged**